fluent.runtime/fluent/runtime/prepare.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import absolute_import, unicode_literals
+from fluent.syntax import ast as FTL
+from . import resolver
+
+
+class Compiler(object):
+    def __init__(self, use_isolating=False):
+        self.use_isolating = use_isolating
+
+    def __call__(self, item):
+        if isinstance(item, FTL.BaseNode):
+            return self.compile(item)
+        if isinstance(item, (tuple, list)):
+            return [self(elem) for elem in item]
+        return item
+
+    def compile(self, node):
+        nodename = type(node).__name__
+        if not hasattr(resolver, nodename):
+            return node
```

Comment:
With this kind of dynamic lookup of attributes on the `resolver` module, I think it would help comprehension if:

1. The resolver module gained an `__all__` attribute for show what it is exporting clearly, including all the `BaseResolver` subclasses.
2. Somewhere next to that there was a brief explanation of how these classes are looked up.

Otherwise someone new to the code base sees the `resolver.Message` class, for instance, but can't find any other references to it or work out how it is actually used.

Reply:
I added a doc string to the module and cleaned up the global namespace a bit more.

I didn't go for adding an `__all__` after all. I started doing it, and then looked at the impact it had on our code paths, and there weren't that I could see. In particular, the pre-evaluator imports the module, and the complete namespace is on the module object. The only thing that would differ is `from fluent.runtime.resolver import *`, which we're not doing. Thus the `__all__ `would mostly be a source of typos and overlooked changes.